northd/northd.c

```diff
@@ -8679,12 +8692,63 @@ build_lswitch_arp_nd_responder_known_ips(struct ovn_port *op,
                     "outport = inport; "
                     "flags.loopback = 1; "
                     "output;",
-                    op->peer->lrp_networks.ea_s,
-                    op->peer->lrp_networks.ea_s);
+                    ea_s,
+                    ea_s);
 
                 ovn_lflow_add_with_hint(lflows, op->od, S_SWITCH_IN_ARP_ND_RSP,
                     50, ds_cstr(match), ds_cstr(actions), &op->nbsp->header_);
             }
+
+            /* Add IPv6 NDP responses.
+             * For ND solicitations, we need to listen for both the
+             * unicast IPv6 address and its all-nodes multicast address,
+             * but always respond with the unicast IPv6 address. */
+            if (proxy_arp_addrs.n_ipv6_addrs) {
+                struct ds ip6_dst_match = DS_EMPTY_INITIALIZER;
+                struct ds nd_target_match = DS_EMPTY_INITIALIZER;
+                for (size_t j = 0; j < proxy_arp_addrs.n_ipv6_addrs; j++) {
+                    ds_put_format(&ip6_dst_match, "%s/%u, %s/%u, ",
```

Comment:
It seems that the string will have `, ` hanging at the end. You can use 
```
ds_chomp(&ip6_dst_match, ' ');
ds_chomp(&ip6_dst_match, ',');
```
or `ds_truncate(&ip6_dst_match, ip6_dst_match.length - 2);`